src/app/shop/[category]/[productId]/_components/TabContents/DetailTab.tsx

```diff
@@ -0,0 +1,85 @@
+'use client';
+
+import { ProductReviewType } from '@/types/ProductReviewTypes';
+import classNames from 'classnames/bind';
+import { RefObject, useCallback, useEffect, useRef, useState } from 'react';
+import DetailImage from './DetailImage';
+import styles from './DetailTab.module.scss';
+import ReturnExchangeInfo from './ReturnExchangeInfo';
+import ProductReviewList from './Review/ProductReviewList';
+
+const cn = classNames.bind(styles);
+
+interface DetailTabProps {
+  detailsImg: string;
+  reviewData: ProductReviewType;
+}
+
+export default function DetailTab({ detailsImg, reviewData }: DetailTabProps) {
+  interface TabData {
+    id: number;
+    button: string;
+    content: JSX.Element;
+    ref: React.RefObject<HTMLDivElement>;
+  }
+  const tabRefs = [useRef<HTMLDivElement>(null), useRef<HTMLDivElement>(null), useRef<HTMLDivElement>(null)];
+
+  const tabData: TabData[] = [
+    { id: 0, button: '상품 상세 정보', content: <DetailImage detailsImg={detailsImg} />, ref: tabRefs[0] },
+    { id: 1, button: '교환 및 반품 안내', content: <ReturnExchangeInfo />, ref: tabRefs[1] },
+    { id: 2, button: '리뷰', content: <ProductReviewList data={reviewData} />, ref: tabRefs[2] },
+  ];
+
+  const [activeTab, setActiveTab] = useState(0);
+
+  const handleTabClick = useCallback((id: number, ref: RefObject<HTMLDivElement>) => {
+    setActiveTab(id);
+    if (ref.current) {
+      window.scrollTo({
+        top: ref.current.offsetTop - 140,
+      });
+    }
+  }, []);
+
+  useEffect(() => {
+    const handleScroll = () => {
+      const scrollPosition = window.scrollY;
+      tabRefs.forEach((tabRef, index) => {
+        if (tabRef.current) {
+          const { offsetTop } = tabRef.current;
+          const offsetBottom = offsetTop + tabRef.current.clientHeight;
+          if (scrollPosition >= offsetTop - 250 && scrollPosition < offsetBottom) {
+            setActiveTab(index);
+          }
+        }
+      });
+    };
+
+    window.addEventListener('scroll', handleScroll);
+    return () => window.removeEventListener('scroll', handleScroll);
+  });
```

Comment:
```suggestion
  useEffect(() => {
    const handleScroll = () => {
      const scrollPosition = window.scrollY;
      tabRefs.forEach((tabRef, index) => {
        if (tabRef.current) {
          const { offsetTop } = tabRef.current;
          const offsetBottom = offsetTop + tabRef.current.clientHeight;
          if (scrollPosition >= offsetTop - 250 && scrollPosition < offsetBottom) {
            setActiveTab(index);
          }
        }
      });
    };

    window.addEventListener('scroll', handleScroll);
    return () => window.removeEventListener('scroll', handleScroll);
  }, []);
```
혹시 이런경우 디펜던시 설정 안해도 괜찮은가요? 궁금

Reply:
**React Hook useEffect has a missing dependency: 'tabRefs'. Either include it or remove the dependency array.**
경고가 떠서 tabRefs를 넣으면  useMemo로 감싸라는 경고가 뜨고 감싸면 useRef를 안에서 쓸 수 없다는 경고가 떠서 일단 지웠는데 되긴했습니다..! 

제 생각에도 빈 배열을 넣는게 맞는거 같은데 그럼 경고를 여기에만 꺼도 될까요??

Reply:
디펜던시를 넣지않으면 해당 훅은 리렌더링마다 매번 실행될거에요 그래서 트리거를 설정하는걸로 알고있어요
종속성으로 tabRefs까지만 넣는게 맞는거같은데
```tsx
  const tabRefs = useRef([useRef<HTMLDivElement>(null), useRef<HTMLDivElement>(null), useRef<HTMLDivElement>(null)]).current;
  ```
  ref를 한번더 잡아서 객체를 참조해보면 어떻게되요?

Reply:
그렇게 하면 forEach를 못 돌려서 이렇ㄱㅔ 하니 해결 됐어요!!

``` js
  const [tabRefs] = useState([
    useRef<HTMLDivElement>(null),
    useRef<HTMLDivElement>(null),
    useRef<HTMLDivElement>(null),
  ]);
```